exp/api/v1alpha4/azuremanagedcontrolplane_webhook.go

```diff
@@ -291,3 +296,147 @@ func (r *AzureManagedControlPlane) validateSSHKey() error {
 
 	return nil
 }
+
+func (r *AzureManagedControlPlane) validateAadProfile() error {
```

Comment:
I see lots of immutable here, so overall, which field is mutable? what kinds of update is allowed?

Reply:
managed can be mutated only to true before and after cluster creation
adminGroupIds can be mutated, but must have minimum one if managed is set to true.

other 4 fields are w.r.t aad and can be disabled only while moving from legacy to managed aks aad support.